dragnet/util.py

```diff
@@ -90,20 +97,75 @@ def evaluation_metrics(predicted, actual, bow=True):
         p = predicted
         a = actual
 
-    if len(p) == 0:
+    try:
+        precision = true_positive / len(p)
+    except ZeroDivisionError:
         precision = 0.0
-    else:
-        precision = true_positive / float(len(p))
-
-    if len(a) == 0:
+    try:
+        recall = true_positive / len(a)
+    except ZeroDivisionError:
         recall = 0.0
-    else:
-        recall = true_positive / float(len(a))
-
-    if precision + recall == 0:
+    try:
+        f1 = 2.0 * (precision * recall) / (precision + recall)
+    except ZeroDivisionError:
         f1 = 0.0
-    else:
-        f1 = 2.0 * precision * recall / (precision + recall)
 
     # return (precision, recall, f1, dameraulevenshtein(predicted, actual))
     return (precision, recall, f1)
+
+
+def get_and_union_features(features):
+    """
+    Get and combine features in a :class:`FeatureUnion`.
+
+    Args:
+        features (str or List[str], ``Features`` or List[``Features``], or List[Tuple[str, ``Features``]]):
+            One or more features to be used to transform blocks into a matrix of
+            numeric values. If more than one, a :class:`FeatureUnion` is
+            automatically constructed. Example inputs::
+
+                features = 'weninger'
+                features = ['weninger', 'kohlschuetter']
+                features = WeningerFeatures()
+                features = [WeningerFeatures(), KohlschuetterFeatures()]
+                features = [('weninger', WeningerFeatures()), ('kohlschuetter', KohlschuetterFeatures())]
+
+    Returns:
+        :class:`FeatureUnion` or ``Features``
+    """
+    if not features:
+        raise ValueError('invalid `features`: may not be null')
+    if isinstance(features, (list, tuple)):
+        if isinstance(features[0], tuple):
+            return FeatureUnion(features)
+        elif isinstance(features[0], string_):
+            return FeatureUnion([(feature, get_feature(feature)) for feature in features])
+        else:
+            return make_union(*features)
+    elif isinstance(features, string_):
+        return get_feature(features)
+    else:
+        return features
+
+
+def load_pickled_model(filename, dirname=None):
+    """
+    Load a pickled ``Extractor`` model from disk.
+
+    Args:
+        filename (str): Name of pickled model file under ``dirname``.
+        dirname (str): Name of directory on disk containing the pickled model.
+            If None, dragnet's default pickled model directory is used:
+            /path/to/dragnet/pickled_models/[PY_VERSION]_[SKLEARN_VERSION]
+
+    Returns:
+        :class:`dragnet.extractor.Extractor`
+    """
+    if dirname is None:
+        if PY2:
+            pkg_dirname = pkgutil.get_loader('dragnet').filename
+        else:
+            pkg_dirname = pkgutil.get_loader('dragnet').name
+        dirname = os.path.join(pkg_dirname, 'pickled_models', model_path)
+    filepath = os.path.join(dirname, filename)
+    return joblib.load(filepath)
```

Comment:
f
You should use pkgutil.get_data function to get the data file directory.
This only gives relative directory and causes file does not exist exception if I try to use the library after installing with the make file.

I changed line 164 to 170 as shown in bellow in my local repository. 
    if dirname is None:
        pkg_name = pkgutil.get_loader('dragnet').name
        pkg_data_path = os.path.join('pickled_models', model_path, filename)
        filepath = pkgutil.get_data(pkg_name, pkg_data_path)
    else:
        filepath = os.path.join(dirname, filename)